powersimdata/input/converter/pypsa_to_grid.py

```diff
@@ -130,33 +129,45 @@ def _get_storage_gen(df, storage_type):
 
 
 class FromPyPSA(AbstractGrid):
-    """Grid builder for PyPSA network object.
-
-    :param pypsa.Network network: Network to read in.
-    :param bool add_pypsa_cols: PyPSA data frames with renamed columns appended to
-        Grid object data frames.
-    """
+    """Grid builder for PyPSA network object."""
 
     def __init__(self, network, add_pypsa_cols=True):
-        """Constructor"""
+        """Constructor."""
         super().__init__()
-        self._read_network(network, add_pypsa_cols=add_pypsa_cols)
+        self.network = network
+        self.add_pypsa_cols = add_pypsa_cols
+
+    def _set_interconnect(self):
+        if self.interconnect is None:
+            self.interconnect = self.network.name.split(", ")
+
+    def _set_data_loc(self):
+        if len(self.interconnect) > 1:
+            self.data_loc = self.interconnect[0]
+        else:
+            self.data_loc = "pypsa"
+
+    def _set_zone_mapping(self):
+        n = self.network
+        if any(self.id2zone) and any(self.zone2id):
+            return
+        # only relevant if the PyPSA network was originally created from PSD
+        if "zone_id" in n.buses and "zone_name" in n.buses:
+            uniques = ~n.buses.zone_id.duplicated() * n.buses.zone_id.notnull()
+            self.zone2id = (
+                n.buses[uniques].set_index("zone_name").zone_id.astype(int).to_dict()
+            )
+            self.id2zone = _invert_dict(self.zone2id)
 
-    def _read_network(self, n, add_pypsa_cols=True):
+    def build(self):
         """PyPSA Network reader.
 
         :param pypsa.Network n: PyPSA network to read in.
         :param bool add_pypsa_cols: PyPSA data frames with renamed columns appended to
             Grid object data frames
```

Comment:
We can remove these parameters from the docstring since those are passed to the constructor. See above comment